### PR TITLE
Update to Glimmer

### PR DESCRIPTION
**Pull request by cibernox**
Since Ember Octane was released in 3.15 (11 minor versions ago!), I think it is reasonable to convert the components in this addon to Glimmer components.

While for users in 3.15 and above this shouldn't be a breaking change, it would be for users on order versions, so we might want to bump the version number after this.

Thoughts?

### CODE REVIEW

**Comment by mehulkar** [comment]
@ronco is this something you'd be ok to merge?

**Comment by mehulkar** [comment]
@rwjblue is this something you could merge/release?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+language: node_js
+node_js:
+  # we recommend testing addons with the same minimum supported node version as Ember CLI
+  # so that your addon works for all apps
+  - "10"
```
Comment (rwjblue) [inline .travis.yml:6]: Since we are going to major, we might as well drop Node 10 too (since it's already EOL'ed)

File: README.md [hunk 2]
```diff
@@ -6,10 +6,7 @@ This addon lets you populate `<head>` tag from your Ember code without any direc
 
 The hope is that, in the future, Ember will provide a mechanism for populating `<head>` tag from your app. Until then, this addon provides that functionality.
 
-
-## Compatibility
-
-* Ember.js v2.18 or above
+* Ember.js v3.16 or above
 * Ember CLI v2.13 or above
 * Node.js v10 or above
```
Comment (rwjblue) [inline README.md:11]:
```suggestion
* Node.js v12 or above
```

File: addon/components/head-layout.js [hunk 3]
```diff
@@ -1,37 +1,30 @@
-import Component from '@ember/component';
+import Component from '@glimmer/component';
 import { inject as service } from '@ember/service';
-import layout from '../templates/components/head-layout';
```
Comment (rwjblue) [inline addon/components/head-layout.js:3]: Let's either use colocated template + component, or bring this import back and use `setComponentTemplate` directly (which is the analogue to setting `layout` property on the prototype).

The second option might look like:

```js
import { setComponentTemplate } from '@ember/component';
import layout from '../templates/components/head-layout';

export default class HeadLayout extends Component {
  // ..snip...
}

setComponentTemplate(layout, HeadLayout);
```

---

Just to note: this is basically how colocated templates work.

File: app/templates/components/head-layout.js [hunk 4]
```diff
@@ -0,0 +1 @@
+export { default } from 'ember-cli-head/templates/components/head-layout';
```
Comment (rwjblue) [inline app/templates/components/head-layout.js:1]: We should either colocate the template (e.g. move it from `addon/templates/components/head-layout.hbs` to `addon/components/head-layout.hbs`) or move to using `setComponentTemplate` in the class definition file (see other comment for exmaple of that).

**Review by rwjblue — CHANGES_REQUESTED** [review]
Looks good overall, I left a couple inline comments to address but should hopefully be pretty minor tweaks.

File: .travis.yml (hunk 1 above)
Comment (cibernox) [reply .travis.yml:6]: good point

**Comment by cibernox** [comment]
@rwjblue I believe it should be good now

File: package.json [hunk 5]
```diff
@@ -20,58 +20,71 @@
...
   "engines": {
-    "node": "10.* || >= 12"
+    "node": "12.* || >= 14"
   },
   "publishConfig": {
     "registry": "https://registry.npmjs.org"
   },
   "resolutions": {
-    "fastboot": "3.1.0"
+    "fastboot": "3.3.0"
```
Comment (rwjblue) [inline package.json:87]:
```suggestion
    "fastboot": "3.1.3"
```

**Comment by cibernox** [comment]
we have both a package-lock.json and a yarn.lock. It's probably my fault I run `npm install`.

**Comment by rwjblue** [comment]
Ya, I've fixed that specific issue (moved to using npm only) and finally got CI to run properly again, but need to spend some time to dig into why things are failing to render as expected.

I _think_ it might be due to [`ec94519` (#94)](https://github.com/ronco/ember-cli-head/pull/94/commits/ec94519e0216c4c81dc6af6d6274de0f489b77d2), but I'm not sure.

**Comment by rwjblue** [comment]
Yippie!!! Looks like that last change finally finished the last of the CI scenarios off.

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Released in https://github.com/ronco/ember-cli-head/releases/tag/v2.0.0

**Pull request merged**